core/json-utils/src/main/java/software/amazon/awssdk/protocols/jsoncore/JsonWriter.java

```diff
@@ -0,0 +1,299 @@
+    public JsonWriter writeValue(Instant instant) {
+        try {
+            generator.writeNumber(DateUtils.formatUnixTimestampInstant(instant));
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(BigDecimal value) {
+        try {
+            /**
+             * Note that this is not how the backend represents BigDecimal types. On the wire
+             * it's normally a JSON number but this causes problems with certain JSON implementations
+             * that parse JSON numbers as floating points automatically. (See API-433)
+             */
+            generator.writeString(value.toString());
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeValue(BigInteger value) {
+        try {
+            generator.writeNumber(value);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    public JsonWriter writeNumber(String number) {
+        try {
+            generator.writeNumber(number);
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+        return this;
+    }
+
+    /**
+     * Closes the generator and flushes to write. Must be called when finished writing JSON
+     * content.
+     */
+    private void close() {
+        try {
+            generator.close();
+        } catch (IOException e) {
+            throw new JsonGenerationException(e);
+        }
+    }
+
+    /**
+     * Get the JSON content as a UTF-8 encoded byte array. It is recommended to hold onto the array
+     * reference rather then making repeated calls to this method as a new array will be created
+     * each time.
+     *
+     * @return Array of UTF-8 encoded bytes that make up the generated JSON.
+     */
+    public byte[] getBytes() {
+        close();
+        return baos.toByteArray();
+    }
+
+    protected software.amazon.awssdk.thirdparty.jackson.core.JsonGenerator getGenerator() {
+        return generator;
+    }
+
+    /**
+     * A builder for configuring and creating {@link JsonWriter}. Created via {@link #builder()}.
+     */
+    public static final class Builder {
+        private JsonFactory jsonFactory = DEFAULT_JSON_FACTORY;
+
+        private Builder() {
+        }
+
+        /**
+         * The {@link JsonFactory} implementation to be used when parsing the input. This allows JSON extensions like CBOR or
+         * Ion to be supported.
+         *
+         * <p>It's highly recommended us use a shared {@code JsonFactory} where possible, so they should be stored statically:
+         * http://wiki.fasterxml.com/JacksonBestPracticesPerformance
+         *
+         * <p>By default, this is {@link #DEFAULT_JSON_FACTORY}.
+         */
+        public JsonWriter.Builder jsonFactory(JsonFactory jsonFactory) {
```

Comment:
Is this API needed?

Reply:
It's provided for the parser, JsonNodeParser, so it's here for consistency.